impl/src/planning.rs

```diff
@@ -248,19 +252,14 @@ impl CrateCatalogEntry {
 impl CrateCatalog {
   /** Produces a CrateCatalog using the package entries from a metadata blob.*/
   pub fn new(metadata: &Metadata) -> Self {
+    let resolve = metadata.resolve.as_ref().unwrap();
```

Comment:
As a practice, I've tried to make sure my `unwraps()` have a justification as a comment above the invocation site. Is this known to be safe, or is there some reason why it's always safe in this case? Same for relow.

Reply:
Rather than justifying them, I've just replaced them with error handlers.

IMO it would be better to propagate an explicit error if "cargo metadata" returned malformed output, rather than panicking.